templates/hooks/vpc_peering_connection/sdk_read_many_post_set_output.go.tpl

```
@@ -1,4 +1,8 @@
 
+	// This prevents reference resolution errors when adopting existing resources where these fields are not provided in the manifest.
+	if ko.Spec.VPCID == nil && ko.Status.RequesterVPCInfo != nil && ko.Status.RequesterVPCInfo.VPCID != nil {
+		ko.Spec.VPCID = ko.Status.RequesterVPCInfo.VPCID
+	}
```

should we return an error that makes sense as else?
It seems like describevpcpeeringconnection does not return requesterVPCInfo if the peering connection is not active...

---

`requester-vpc-info.vpc-id `field is always returned in the `DescribeVpcPeeringConnections` API response, regardless of the peering connection's status. Even for deleted etc.

the `requester-vpc-info.cidrBlock` is not returned.

something like this:
```
  conditions:
  - lastTransitionTime: "2025-04-14T02:24:03Z"
    message: Resource synced successfully
    reason: ""
    status: "True"
    type: ACK.ResourceSynced
  requesterVPCInfo:
    ownerID: "xxxxx"
    region: us-west-2
    vpcID: vpc-0....
  status:
    code: deleted
```